plugins/transport-grpc/src/test/java/org/opensearch/transport/grpc/NettyGrpcClient.java

```diff
@@ -0,0 +1,171 @@
+            }
+
+            @Override
+            public void onCompleted() {
+                latch.countDown();
+            }
+        };
+
+        StreamObserver<ServerReflectionRequest> requestObserver = reflectionStub.serverReflectionInfo(responseObserver);
+        requestObserver.onNext(ServerReflectionRequest.newBuilder().setListServices("").build());
+        requestObserver.onCompleted();
+
+        try {
+            if (!latch.await(5, TimeUnit.SECONDS)) {
+                throw new RuntimeException(NettyGrpcClient.class.getSimpleName() + " timed out waiting for response.");
+            }
+        } catch (InterruptedException e) {
+            throw new RuntimeException(NettyGrpcClient.class.getSimpleName() + " interrupted waiting for response: " + e.getMessage());
+        }
+
+        return respServices;
+    }
+
+    /**
+     * Request server status.
+     * @return HealthCheckResponse.ServingStatus.
+     */
+    public HealthCheckResponse.ServingStatus checkHealth() {
+        return healthStub.check(HealthCheckRequest.newBuilder().build()).getStatus();
+    }
+
+    public static class Builder {
+        private Boolean mTLS = false;
+        private Boolean insecure = false;
+        private TransportAddress addr;
+
+        private static final ApplicationProtocolConfig CLIENT_ALPN = new ApplicationProtocolConfig(
+            ApplicationProtocolConfig.Protocol.ALPN,
+            ApplicationProtocolConfig.SelectorFailureBehavior.NO_ADVERTISE,
+            ApplicationProtocolConfig.SelectedListenerFailureBehavior.ACCEPT,
+            ApplicationProtocolNames.HTTP_2
+        );
+
+        Builder() {}
+
+        public NettyGrpcClient build() throws SSLException {
+            NettyChannelBuilder channelBuilder = NettyChannelBuilder.forAddress(addr.getAddress(), addr.getPort())
+                .proxyDetector(NOOP_PROXY_DETECTOR);
+
+            if (mTLS) {
+                SslContextBuilder builder = SslContextBuilder.forClient();
+                builder.applicationProtocolConfig(CLIENT_ALPN);
+                builder.keyManager(getTestKeyManagerFactory(CLIENT_KEYSTORE));
+                builder.trustManager(InsecureTrustManagerFactory.INSTANCE);
+                channelBuilder.sslContext(builder.build());
+            } else if (insecure) {
+                SslContextBuilder builder = SslContextBuilder.forClient();
+                builder.applicationProtocolConfig(CLIENT_ALPN);
+                builder.trustManager(InsecureTrustManagerFactory.INSTANCE);
+                channelBuilder.sslContext(builder.build());
+            } else {
+                channelBuilder.usePlaintext();
+            }
+
+            return new NettyGrpcClient(channelBuilder);
+        }
+
+        public Builder setAddress(TransportAddress addr) {
+            this.addr = addr;
+            return this;
+        }
+
+        /**
+         * Enable mTLS - load client keystore.
+         */
+        public Builder mTLS(boolean enable) {
```

Comment:
I would suggest to name it client auth (since mTLS basically covers client/server communication but we are only able to manage the client part here):
```suggestion
        public Builder clientAuth(boolean enable) {
```